src/sss_client/subid/sss_subid.c

```diff
@@ -87,11 +87,20 @@ enum subid_status shadow_subid_list_owner_ranges(const char *user,
     }
 
     SAFEALIGN_COPY_UINT32(&num_results, repbuf, NULL);
+    if (num_results > (replen/sizeof(uint32_t) - 2)/2) {
```

Comment:
Hi,

I would suggest to check first if `replen` has a suitable value. Size all variables are unsigned it might be if `replen == sizeof(uint32_t)` you end up with negative values on the right hand side which might be larger in the comparison.

Btw, gcc has some built-in functions to check for integer overflows, https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html.

bye,
Sumit

Reply:
Thank you. Added.

Reply:
Thanks,

I know I'm nitpicking but it looks it is expected that replen is an even number. An additional  ` (replen % 2)` would make sure that the integer division by 2 always returns the expected integer.

bye,
Sumit

Reply:
Updated.